api/src/main/java/org/openmrs/Diagnosis.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * Diagnosis class defines the identification of the nature of an illness or other problem 
+ * by examination of the symptoms during an encounter (visit or interaction of a patient with a healthcare worker).
+ **/
+
+public class Diagnosis extends BaseChangeableOpenmrsData{
+	
+	private Integer diagnosisId;
+	
+	private Encounter encounter;
+	
+	private CodedOrFreeText diagnosis;
+	
+	private Condition condition;
+
+	private ConditionVerificationStatus certainty;
+	
+	private Integer rank;
+	
+	public Diagnosis() {
+	}
+
+	/**
+	 * Diagnosis constructor with encounter, CodedOrFreeText, id, ConditionVerificationStatus, and rank
+	 *
+	 * @param encounter the encounter during which the diagnosis was made
+	 * @param diagnosis the CodedOrFreeText of the diagnosis
+	 * @param certainty the ConditionVerificationStatus of the diagnosis
+	 * @param rank the rank of the diagnosis
+	 */
+	
+	public Diagnosis(Encounter encounter, CodedOrFreeText diagnosis, ConditionVerificationStatus certainty, Integer rank) {
+		this.encounter = encounter;
+		this.diagnosis = diagnosis;
+		this.certainty = certainty;
+		this.rank = rank;
+	}
+	
+	/**
+	 * Returns the diagnosisId.
+	 * 
+	 * @return diagnosisId
+	 */
+	@Override
+	public Integer getId() {
+		return diagnosisId;
+	}
+
+	/**
+	 * Sets diagnosis diagnosisId
+	 * 
+	 * @param diagnosisId
+	 */
+	@Override
+	public void setId(Integer diagnosisId) {
+		this.diagnosisId = diagnosisId;
+	}
+
+	/**
+	 * Returns the encounter associated with this diagnosis.
```

Comment:
Did you keenly observe how the javadocs are written in this commit? https://github.com/openmrs/openmrs-core/commit/f5132a1dda4a6a8fcbfe06a32dcbbe4b7a0a270d